packages/serverless/src/domain/use-case/permission-security.ts

```diff
@@ -44,23 +48,27 @@ export class PermissionSecurity {
 
   // Get permission of a collection
   public static async collection(
-    databaseName: string,
-    collectionName: string,
-    actor: string,
+    paramCollection: TPermissionSudo<TParamCollection>,
     session?: ClientSession
   ): Promise<PermissionBase> {
+    const { databaseName, collectionName, actor, sudo } = paramCollection;
     // If user is database owner then return all system permissions
     if (await isDatabaseOwner(databaseName, actor)) {
       return PermissionBase.permissionAll();
     }
     const imMetadataCollection =
       ModelMetadataCollection.getInstance(databaseName);
-    const metadata = await imMetadataCollection.findOne(
-      {
-        collectionName,
-      },
-      { session }
-    );
+    // If ownership and permission is not provided then fetch from metadata collection
+    const metadata =
+      sudo ||
+      (
+        await imMetadataCollection.findOne(
```

Comment:
We should throw an error if the `findOne` operation on the metadata collection returns undefined, rather than proceeding. Every collection is required to have its own metadata; if it doesn't, that's an indication of an issue.